Source/Core/Core/MemTools.cpp

```diff
@@ -60,11 +62,28 @@ static LONG NTAPI Handler(PEXCEPTION_POINTERS pPtrs)
     // virtual address of the inaccessible data
     uintptr_t fault_address = (uintptr_t)pPtrs->ExceptionRecord->ExceptionInformation[1];
     SContext* ctx = pPtrs->ContextRecord;
-
-    if (Core::System::GetInstance().GetJitInterface().HandleFault(fault_address, ctx))
+    Core::System& system = Core::System::GetInstance();
+    Memory::MemoryManager& memory = system.GetMemory();
+    if (system.GetJitInterface().HandleFault(fault_address, ctx))
```

Comment:
Wouldn't it make more sense to check for non-dirty pages before checking for a fastmem fault? The way you're currently doing it, we have to do fastmem backpatching every time a non-dirty page is written to, which feels unnecessary to me.